### PR TITLE
Polio 550 drag and drop component

### PR DESCRIPTION
**Pull request by olethanh**
The exiting `FileComponent` allowed drag and dropping files, but it wasn't clear to the user.

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [X] Did I add translations

## Changes

- Created `SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`  to replace `ConfirmCancelDialogComponent` and be able to make other types of modals (by composing `SimpleModal`). The main point was to remove `renderTrigger` so that when we conditionally render the modal component, the API calls are not made unless the modal is rendered.
- Created a `makeFullModal` HOC to facilitate making modals with `ConfirmCancelModal`
- Created `CustomInput` based on `OrgUnitTreeviewPicker`
- Added dependency to [react-dropzone](https://react-dropzone.js.org)
- Used `CustomInput` and react-dropzone to make `FileUpload` component: users can drag and drop files into the component. OnDragOver, it will show a dropzone. Users can also click on the input to open the file selection window
- Refactor `CreateEditBudgetEvent` to use `ConfirmCancelModal`, `makFullModal` and `FileUpload`

## How to test

- Go to a campaign's budget
- Click add step
- click on the "Files" input to open file selection window
- then drag and drop another selection of files in the component

To test the refactor:
- Open the modal to create a new step
- Fill it in
- Block `/api/polio/budgetfiles` and `/api/polio/confirm_budget`
- Confirm
- The table should show an additional icon
- click on it
- You should be able to resend files (but you have to re-authorize the endpoints first for it to work)

## Print screen / video

https://user-images.githubusercontent.com/38907762/189665826-73b498b2-ef85-4fc5-beb5-53a4336fa600.mov

## Notes

`SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`,`makeFullModal` and `FileUpload` should be moved to bluesquare-components. I suggest to create a ticket after this PR is merged

### CODE REVIEW

File: hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import React, { FunctionComponent, ReactNode } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import classnames from 'classnames';
+// @ts-ignore
+import { FormControl } from 'bluesquare-components';
+import { Paper, InputLabel, Box } from '@material-ui/core';
+
+const childrenStyle = theme => ({
+    // replicated from .MuiTypography-body1 (except font-family and letter spacing)
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:9]: so why not using typography directly ?

File: hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+const styles = theme => ({
+    placeholder: {
+        alignItems: 'center',
+        fontSize: '16px',
+        flex: '1',
+        marginLeft: '14px',
+        cursor: 'default',
+        color: 'transparent',
+        paddingTop: 0.5,
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:31]: not sure what will be the result of this 0.5 px ??

File: hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    const additionalPaperStyle = icon
+        ? classes.paperWithIcon
+        : 'MuiOutlinedInput-multiline';
+
+    return (
+        <FormControl errors={errors}>
+            <InputLabel
+                shrink={Boolean(children)}
+                required={required}
+                className={`${classnames(
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:123]: ❤️

File: hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                    Boolean(children) && classes.shrinkInputLabel,
+                    'input-label',
+                )} ${errorLabelStyle}`}
+            >
+                {placeholder}
+            </InputLabel>
+            <Paper
+                variant="outlined"
+                elevation={0}
+                className={`${classes.paper} ${enabledStyle} ${errorStyle} ${additionalPaperStyle}`}
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx]: 😠

File: hat/assets/js/apps/Iaso/components/DragAndDrop/FilesUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+type Props = {
+    multi?: boolean;
+    // eslint-disable-next-line no-unused-vars
+    onFilesSelect: (files: File[]) => void;
+    files: File[];
+    placeholder?: string;
+};
+
+const Icon = (
+    <IconButton
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/components/DragAndDrop/FilesUpload.tsx]: So this is not a button, maybe use only ToolTip and the icon

File: plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx [hunk 6]
```diff
@@ -70,7 +78,7 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
                         onSuccess: () => {
                             finalize(result.id, {
                                 onSuccess: () => {
-                                    closeModal.closeDialog();
```
Comment (beygorghor) [inline plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx:73]: 👍

File: plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx [hunk 7]
```diff
@@ -208,7 +210,11 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
                 maxWidth="sm"
                 cancelMessage={MESSAGES.cancel}
                 confirmMessage={MESSAGES.send}
-                renderTrigger={renderTrigger}
```
Comment (beygorghor) [inline plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx:211]: 🥳

File: plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx [hunk 8]
```diff
@@ -270,14 +276,21 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
...
                         <Box mt={2}>
-                            <FileInputComponent
-                                keyValue="files"
-                                required={currentType === 'edit'}
-                                multiple
-                                onChange={onChange}
-                                value={values.files}
-                                errors={getErrors('files')}
-                                label={MESSAGES.filesUpload}
+                            {/* <FileInputComponent
```
Comment (beygorghor) [inline plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx]: we can remove this then

**Review by beygorghor — APPROVED** [review]
Nice job ! only small remarks

File: hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx (hunk 1 above)
Comment (quang-le) [reply hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:9]: Because I needed to add the padding manually if I recall correctly

**Pull request merged**